src/menus/PlayerSelect.java

```diff
@@ -0,0 +1,203 @@
+package menus;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import utils.Gamepad;
+import DLibX.*;
+
+import java.awt.Color;
+import java.awt.Font;
+
+public class PlayerSelect extends Menu {
+
+    private ArrayList<Gamepad> players = new ArrayList<Gamepad>(Arrays.asList(null, null, null, null));
+
+    public PlayerSelect(DConsole dc, ArrayList<Gamepad> gamepads) {
+        super(dc, gamepads);
+    }
+
+    /**
+     * Runs the player selection screeen
+     * 
+     */
+    public void play() {
+        boolean exitMenu = false;
+
+        while (!exitMenu) {
+
+            drawGui();
+            
+            for (Gamepad gamepad : this.gamepads) {
+                gamepad.poll();
+               
+                // allow players to select a color (player number)
+                if (gamepad.getBButton() && players.get(0) == null && !players.contains(gamepad)) {
+                    players.set(0, gamepad);;
+                } else if (gamepad.getXButton() && players.get(1) == null && !players.contains(gamepad)) {
+                    players.set(1, gamepad);
+                } else if (gamepad.getAButton() && players.get(2) == null && !players.contains(gamepad)) {
+                    players.set(2, gamepad);
+                } else if (gamepad.getYButton() && players.get(3) == null && !players.contains(gamepad)) {
+                    players.set(3, gamepad);
+                }
+                
+            }
+
+            // allow a player to cancel their choice
+            for (Gamepad player : players) {
+                if (player != null) {
+                    if (player.getBackButton()) {
+                        int index = players.indexOf(player);
+                        if (index != -1) {
+                            players.set(index, null);
+                        }
+                    }
+                }
+            }
+            
+            // break loop to next screen
+            if (
+                players.get(0) != null &&
+                players.get(1) != null &&
+                players.get(2) != null &&
+                players.get(3) != null &&
+                players.get(0).getStartButton()
+            ) {
+                exitMenu = true;
+                System.out.println(players);
+            }
+        }    
+    }
+
+    /**
+     * final player assignments
+     * 
+     * @return list of final player assignments (Arraylist<Gamepad>)
+     */
+    public ArrayList<Gamepad> getPlayers() {
+        return this.players;
+    }
+
+    // draw static GUI elements
+    private void drawGui() {
+
+        // fonts
+        Font arialTitle = new Font("arial", 1, 80);
+        Font arialSmall = new Font("arial", 1,30);
+        Font arialMedium = new Font("arial", 1, 40);
+        Font arialVerySmall = new Font("arial", 1, 20);
```

Comment:
Not important, however 2 things

1. I wonder if we can use a more unique font for this, Arial just looks boring
2. We should move these font styles to some static area so that we can standardize the text throughout the entire app.